tools/src/main/java/org/apache/kafka/tools/ProducerPerformance.java

```diff
@@ -223,9 +232,32 @@ private static ArgumentParser argParser() {
                 .dest("producerConfigFile")
                 .help("producer config properties file.");
 
+        parser.addArgument("--print.metrics")
+                .action(storeTrue())
+                .type(Boolean.class)
+                .metavar("PRINT-METRICS")
+                .dest("printMetrics")
+                .help("print out metrics at the end of the test.");
+
         return parser;
     }
 
+    private static void printMetrics(KafkaProducer producer) {
+        Map<MetricName, ? extends Metric> metrics = producer.metrics();
+        int maxLengthOfDisplayName = 0;
+        for (Metric metric : metrics.values()) {
+            MetricName mName = metric.metricName();
+            String mergedName = mName.group() + ":" + mName.name() + ":" + mName.tags();
+            maxLengthOfDisplayName = maxLengthOfDisplayName < mergedName.length() ? mergedName.length() : maxLengthOfDisplayName;
+        }
+        String outputFormat = "%-" + maxLengthOfDisplayName + "s : %.3f";
+        System.out.println(String.format("\n%-" + maxLengthOfDisplayName + "s   %s", "Metric Name", "Value"));
+        for (Metric metric : metrics.values()) {
+            MetricName mName = metric.metricName();
+            System.out.println(String.format(outputFormat, mName.group() + ":" + mName.name() + ":" + mName.tags(), metric.value()));
+        }
```

Comment:
Perhaps we can pull this code into a ToolsUtils.printMetrics(Map<MetricName, ? extends Metric> metrics) method and share between the consumerPerf and producerPerf. It would also be useful to print out the metrics in alphabetical order.